File: Python/queues.py
class PQ:
    def __init__(self):
        self.collection = [[2, 7]]

    def dequeue(self):
        return self.collection.pop()

    def peak(self):
        print(self.collection[len(self.collection) - 1])

    def is_empty(self):
        return (self.collection == 0)

    def enqueue(self, item):
        priority, num = item
        if len(self.collection) == 0:
            self.collection.append([priority, num])
        else:
            for index, element in enumerate(self.collection):
                element_added = False
                coll_prior, coll_num = element
                if coll_prior > priority:
                    self.collection.insert(index, [priority, num])
                    element_added = True
                    break
            if element_added == False:
                self.collection.append([priority, num])


priority_queue = PQ()
print(priority_queue.collection)
priority_queue.peak()
print(priority_queue.is_empty())
print(priority_queue.dequeue())
priority_queue.enqueue([6, 3])
priority_queue.enqueue([8, 3])
priority_queue.enqueue([2, 3])
priority_queue.enqueue([10, 3])
print(priority_queue.collection)

File: JavaScript/queues.js
//STANDARD QUEUE
class Queue {
    constructor(){
        this.collection = [];
    }

    print(){
        return this.collection
    }

    enqueue(value){
        this.collection.push(value);
    }

    dequeue(){
        this.collection.shift();
    }

    front(){
        return this.collection[0];
    }

    size(){
        return this.collection.length;
    }

    isEmpty(){
        return (this.collection.length == 0);
    }
}

// const myQueue = new Queue
// myQueue.enqueue(6)
// myQueue.enqueue(099)
// myQueue.enqueue(6282)
// myQueue.enqueue(8182)
// myQueue.dequeue()
// console.log(myQueue.size())
// console.log(myQueue.front())
// console.log(myQueue.isEmpty())


//PRIORITY QUEUE
class PriorityQueue {
    constructor(){
        this.collection = [];
    }

    print(){
        return this.collection
    }

    enqueue(value){
        if ( this.collection.length === 0){
            this.collection.push(value);
        } else {
            let elementAdded = false;
            for ( let i = 0; i < this.collection.length; i++){
                if (value[1] < this.collection[i][1]){
                    elementAdded = true;
                    break;
                }
            }
            if (elementAdded == false){
                this.collection.push(value);
            }
        }
        
    }

    dequeue(){
        this.collection.shift();
    }

    front(){
        return this.collection[0];
    }

    size(){
        return this.collection.length;
    }

    isEmpty(){
        return (this.collection.length == 0);
    }
}

const myPQ = new PriorityQueue
myPQ.enqueue(['kenke', 8])
myPQ.enqueue(['armin', 4])
myPQ.enqueue(['cole', 10])
myPQ.enqueue(['sarah', 2])
myPQ.enqueue(['harriet', 8])
myPQ.enqueue(['ruth', 6])
console.log(myPQ.print())